mgz/fast/__init__.py

```diff
@@ -178,8 +180,21 @@ def parse_action(action_type, data):
         object_id, = struct.unpack_from('<3xI', data)
         return dict(object_ids=[object_id])
     if action_type == Action.WALL:
-        player_id, x, y = struct.unpack_from('<x3b', data)
-        return dict(player_id=player_id, x=x, y=y)
+        selection_count, player_id, x_start, y_start, x_end, y_end, z0, building_id, z1, const = struct.unpack_from(
```

Comment:
Parse like `selection_count, player_id, x_start, y_start, x_end, y_end, building_id = struct.unpack_from('<6bxh')` and then we don't have to name unused bytes

Reply:
Tested this again with .aoe2recording and .mgz tess files and I'm a bit confused now, the data format in  DE recs seems to be different than in mgz files, e.g:

from 'tests/recs/small.mgz'
sel.   p  x1. y1. x2.  y2.        (building_id)
04   02 3b 2d 3b  2d   00       48 00         00 00 ff ff ff ff 0b 08 00 00   0f 08 00 00   31 08 00 00   33 08 00 00'

from 'tests/recs/de-13.34.aoe2record'
s?  p           x1        y1       x2       y2       0       (building_id)
01 01    00 2d 00 28 00 32 00  1f      00            48 00          0000ffffffff00010000a5220000'

all coordinates are padded with an int8 0. I'm not sure what the purpose of this.

I assume it's for larger map size coordinates which can't fit into int8? (but the coordinates themselves are still int8 unless we assume that the 0 before building id is part of y2 and shift them all to right).

All the other action types (build, attack, move) use floats for coordinates so this seems to be specific to WALL.

Reply:
I guess the best solution would be to  have two different format string based on data length?